python/dgl/transform.py

```diff
@@ -1255,56 +1255,139 @@ def out_subgraph(g, nodes):
+    to each type as a separate graph and compute the
+    `simple` for each of them.
+
+    Writeback mappings can be returned for each edge
+    type subgraph. A writeback mapping is a tensor
+    recording the edge ids in the `simpled` graph for
+    each edges of the original graph. If the graph has
+    only one edge type, a single tensor is returned.
+    If the graph has multiple edge types, a dictionary
+    of tensor is returned using canonical edge types
+    as the key.
+
+    Given a :class:`DGLGraph` object, we return another
+    :class:`DGLGraph` object representing its simplified version.
 
-    TODO(xiangsx): Don't save writeback_mapping into g, but put it into return value.
 
     Parameters
     ----------
-    g : DGLHeteroGraph
-        The heterogeneous graph
+    g : DGLGraph
+        The input graph.
     return_counts : str, optional
-        If given, the returned graph would have a column with the same name that stores
-        the number of duplicated edges from the original graph.
-    writeback_mapping : str, optional
-        If given, the mapping from the edge IDs of original graph to those of the returned
-        graph would be written into edge feature with this name in the original graph for
-        each edge type.
+        If given, the returned graph would have a column with
+        the same name that stores the number of duplicated edges
+        from the original graph.
+    writeback_mapping: bool, optional
+        If True, a write back mapping is returned for each edge
+        type subgraph. If False, only the g is returned.
+        (Default: Ture)
+    share_ndata: bool, optional
+        If True, the node features of the reversed graph will
+        be the same as the original graph. If False, the reversed
+        graph will not have any node features.
+        (Default: True)
 
     Returns
     -------
-    DGLHeteroGraph
-        The new heterogeneous simple graph.
+    DGLGraph
+        A simple graph.
+    F.tensor or dict of F.tensor
+        The writeback mapping. If the graph has only
+        one edge type, a tensor is returned. If the
+        graph has multiple edge types, a dictionary
+        of tensor is return.
 
     Examples
     --------
-    Consider the following graph
-    >>> g = dgl.graph(([0, 1, 2, 1, 1, 1], [1, 3, 2, 3, 4, 4]))
-    >>> sg = dgl.to_simple(g, return_counts='weights', writeback_mapping='new_eid')
-
-    The returned graph would have duplicate edges connecting (1, 3) and (1, 4) removed:
-    >>> sg.all_edges(form='uv', order='eid')
-    (tensor([0, 1, 1, 2]), tensor([1, 3, 4, 2]))
+    **Homographs or Heterographs with A Single Edge Type**
 
-    If ``return_counts`` is set, the returned graph will also return how many edges
-    in the original graph are connecting the endpoints of the edges in the new graph:
-    >>> sg.edata['weights']
-    tensor([1, 2, 2, 1])
+    Create a graph to simple. In the original graph, there are multiple edges between 1 and 2
```

Comment:
"Create a graph to simple" -> "Create a graph for demonstrating `to_simple` API"